testing/deployer/topology/compile.go

```diff
@@ -90,22 +100,17 @@ func compile(logger hclog.Logger, raw *Config, prev *Topology) (*Topology, error
 		nextIndex int // use a global index so any shared networks work properly with assignments
 	)
 
-	foundPeerNames := make(map[string]map[string]struct{})
-	for _, c := range raw.Clusters {
+	compileCluster := func(c *Cluster) (map[string]struct{}, error) {
```

Comment:
A follow-on PR can extract these anonymous methods as real methods. I did it this way to make it more obvious where the bugfixes were, and to add error logging context.